src/inline_table.rs

```diff
@@ -93,6 +93,46 @@ impl InlineTable {
         }
     }
 
+    /// Sort Key/Value Pairs of the table using the using the comparison function `compare`.
+    ///
+    /// The comparison function receives two key and value pairs to compare (you can sort by keys or
+    /// values or their combination as needed).
+    pub fn sort_by<F>(&mut self, compare: F)
+    where
+        F: Fn(&Key, &Item, &Key, &Item) -> std::cmp::Ordering,
```

Comment:
`Item` or `Value`?

Reply:
We're ultimately sorting on a `TableKeyValue`, which has a `Key` and an `Item`. That's why I went with `Item`.

Is there a guarantee that the `Item` will necessarily always be a `Value`? If so, we can sort on `val.value.as_value().unwrap()` instead of `&val.value`.

Your call.

Reply:
Only through developers abusing corners in our API do we allow an `InlineTable` to have an item.  We then pretty much ignore any non-`Value` `Item` in the table (we won't render it).

I wouldn't go as far as `.unwrap()` but I would choose an arbitrary sort policy for it.